### PR TITLE
Uclid benchmarks

### PR DESCRIPTION
**Pull request by polgreen**

### CODE REVIEW

**Comment by polgreen** [comment]
I'm not sure what the error is here. The CI says:
~~~
Checking "non-incremental/AUFDTLIA/uhb_examples/uhb_quantifiers.smt2"
                  File: non-incremental/AUFDTLIA/uhb_examples/uhb_quantifiers.smt2
         Logic in path: AUFDTLIA
          Type in path: non-incremental
         Logic in file: AUFDTLIA
              Encoding: us-ascii
               SMT-LIB: 2.6
               License: CC
              Category: industrial
          Type of file: non-incremental
  Error: Process completed with exit code 1.
~~~
Not sure why it's completed with exit code 1?

**Comment by mpreiner** [comment]
The benchmark submission does not follow the directory structure described in the [submission guidelines](https://github.com/SMT-LIB/benchmark-submission#folder-structure).

File: non-incremental/AUFDTLIA/randybryant/run-btfnt-idata-fullalu-umem-coninv-f5.ucl-property_correspondence-vobj-0001.smt2 [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+     
+     ))
+(assert (forall ((r _enum_0))
+   (and (or (not (= r RNONE)) (= (select initial_238___ucld_1673_RF_var r) 0))
+          (or (= r RNONE)
+              (= (select initial_238___ucld_1673_RF_var r)
+                 (const___ucld_1431_rf0_common r))))
+     
+     ))
+(assert false)
```
Comment (mpreiner) [inline non-incremental/AUFDTLIA/randybryant/run-btfnt-idata-fullalu-umem-coninv-f5.ucl-property_correspondence-vobj-0001.smt2]: This benchmark is trivially unsat. Is this intended? If yes, I see no reason to submit this kind of benchmarks.

Comment (polgreen) [reply non-incremental/AUFDTLIA/randybryant/run-btfnt-idata-fullalu-umem-coninv-f5.ucl-property_correspondence-vobj-0001.smt2]: only for completeness really; this is a set of benchmarks generated by uclid5 from the Randy Bryant paper referenced above. I can remove it

**Comment by polgreen** [comment]
> The benchmark submission does not follow the directory structure described in the [submission guidelines](https://github.com/SMT-LIB/benchmark-submission#folder-structure).

The guidelines say the directory structure should be `[non-]incremental/<logic>/<set-name>/.../<name>.smt2`

The directory structure that has thrown up the error is `non-incremental/AUFDTLIA/uhb_examples/uhb_quantifiers.smt2`

I can't figure out why this doesn't comply? The logic is AUFDTLIA, the set name is uhb_examples

**Comment by mpreiner** [comment]
```
The <set-name> is either <date>-<application>-<submitter>, <date>-<application>, or <date>-<submitter>.
The <date> is >written as YYYYMMDD.
```

Try adding a date.

File: non-incremental/AUFDTLIA/randybryant/run-btfnt-idata-fullalu-umem-coninv-f5.ucl-property_correspondence-vobj-0001.smt2 (hunk 1 above)
Comment (mpreiner) [reply non-incremental/AUFDTLIA/randybryant/run-btfnt-idata-fullalu-umem-coninv-f5.ucl-property_correspondence-vobj-0001.smt2]: Please remove all benchmarks that include `(assert false)`.

**Comment by mpreiner** [comment]
@polgreen Can you please fix the remaining issues reported by the Dolmen check?

**Comment by polgreen** [comment]
I think the remaining errors are spurious? 

They are all similar to this:

> File "non-incremental/AUFDTLIA/20210304-randybryant/run-btfnt-idata-fullalu-amem-coninv-f5.ucl-property_correspondence-vobj-0002.smt2", line 991, character 23-36:
Warning This is a non-linear expression according to the smtlib spec.
        Hint: multiplication in strict linear arithmetic expects an integer
          or rational literal and a symbol (variable or constant) but was
          given:
          - an integer coefficient
          - an arbitrary expression with top symbol not in Arithmetic

Line 991 is:
`(+ a!2 (* (- 1) a!3))`

So `-1` is an integer, and `a!3` is set in the following let expression:

~~~
(assert (let ((a!1 (ite (or (= const___ucld_1491_E_icode0_pipe_common IRMMOVQ)
                    (= const___ucld_1491_E_icode0_pipe_common IMRMOVQ)
                    (= const___ucld_1491_E_icode0_pipe_common IIRMOVQ)
                    (= const___ucld_1491_E_icode0_pipe_common IJXX))
                const___ucld_1497_E_valC0_pipe_common
                (ite (or (= const___ucld_1491_E_icode0_pipe_common ICALL)
                         (= const___ucld_1491_E_icode0_pipe_common IPUSHQ))
                     (- 8)
                     8)))
      (a!3 (ite (or (= const___ucld_1491_E_icode0_pipe_common IRET)
                    (= const___ucld_1491_E_icode0_pipe_common IOPQ)
                    (= const___ucld_1491_E_icode0_pipe_common IRMMOVQ)
                    (= const___ucld_1491_E_icode0_pipe_common IMRMOVQ)
                    (= const___ucld_1491_E_icode0_pipe_common ICALL)
                    (= const___ucld_1491_E_icode0_pipe_common IPUSHQ)
                    (= const___ucld_1491_E_icode0_pipe_common IPOPQ))
                const___ucld_1505_E_valB0_pipe_common
                0)))
~~~
So it is either `0` or `const___ucld_1505_E_valB0_pipe_common`. The latter of these is declared as an integer:

~~~
(declare-fun const___ucld_1505_E_valB0_pipe_common () Int)
~~~

So I think we have an integer and a (variable) symbol, so it should fall into linear arithmetic? I can switch them to the non-linear to make the CI pass but that doesn't seem to be the actual correct logic?

**Comment by mpreiner** [comment]
@polgreen That's fine, we disabled the strict arithmetic checks in Dolmen, which reports theses cases as warnings instead of errors. Thanks for the fixes! Looks like we are good to go.

**Pull request merged**